### DISCUSSION TITLE
Build cache producing error with Turborepo? routes-manifest.json not found.

### DISCUSSION THREAD

**Discussion by florianmartens** [vercel/vercel#7846] [Help] [answered]
Hi,

I'm currently experiencing some issues deploying a Next.js + Turborepo app on Vercel. The initial build ran fine but after pushing an update builds starting to fail with the following error:

```
Tasks:    3 successful, 3 total
--
...
23:17:27.189 | Error: The file "/vercel/path0/apps/app/.next/routes-manifest.json" couldn't be found. This is normally caused by a misconfiguration in your project.
```

I'm setting the project root correctly as `apps/app` - which is the folder that contains my `.next` build folder. 

I'm using the Next.js defaults and am overwriting the build command as: `cd ../.. && yarn turbo run build --scope=app`.

When building locally, I can see that the `routes.manifest.json` is created:

![image](https://user-images.githubusercontent.com/48451846/169618540-b7c1f629-38c9-4b9a-adce-2bdc1bd9ac52.png)

When using the remote build cache, Vercel is not able to locate the file.

I believe the build cache is responsible for the error since I'm able to get builds to complete when forcing turorepo not to use cache with the following command: `cd ../.. && yarn turbo run build --scope=app --force`

Is anyone experiencing the same?

**Comment by jaredcbaum** [comment]
As a temporary solution, I put this for the install command: `npm install --prefix=../.. && npm run build`
This could be problematic because it builds everything in the turborepo

**Comment by jaredcbaum** [comment, marked as answer]
I found the problem for me. The `name` field in the package.json file has to be the same as the folder name in the `apps` folder of the monorepo. In your case @florianmartens, you would have to change the `name` field in `apps/app/package.json` to `app`. Also, I believe your build command should be: `cd ../.. && yarn turbo run build --scope=app...`, note the three dots at the end.

**Reply by jaredcbaum** [reply]
Note: I am using npm workspaces

**Reply by D-Pagey** [reply]
I had same problem. Ensuring the folders were the same as `name` field fixed it for me

**Reply by aceaspades-worldspark** [reply]
Can this answer here **please** get added to the [routes.manifest errors page](https://github.com/vercel/vercel/blob/main/errors/now-next-routes-manifest.md )? I have spent too many hours trying to fix this issue, but this answer was correct for me.

**Reply by inf1nite-lo0p** [reply]
FYI: https://stackoverflow.com/a/75774960/6622233

**Reply by zhawtof** [reply]
Definitely check out what @MohammadxAli mentioned here (I only needed (4). The (3) instructions weren't particularly handy)

Doing all of the above helps but it isn't the full step.

**Comment by miguelgargallo** [comment]
Be careful and make sure you put the same name to json into the folder box

**Reply by neotan** [reply]
> Be careful and make sure you put the same name to json into the folder box

On top of @miguelgargallo 's reply, in case someone ended up landing here for a solution who is using `Turborepo` to manage your multi-apps/packages and deploying them to Vercel. Your Vercel project named `mywebsite.com` needs to be aligned with `package.json`'s `name: mywebsite.com`. Or you may have to override the **Build Command** with the proper `package.json`'s `name`
